bundles/org.openhab.binding.semsportal/src/main/java/org/openhab/binding/semsportal/internal/SEMSPortalHandler.java

```diff
@@ -0,0 +1,268 @@
+    private SEMSPortalConfiguration config = new SEMSPortalConfiguration();
+    private boolean loggedIn;
+    private boolean stationOnline = false;
+    private SEMSToken sessionToken = SESSIONLESS_TOKEN;// gets the default, it is needed for the login
+    private @Nullable SEMSStatus currentStatus;
+
+    private LocalDateTime lastUpdate = LocalDateTime.MIN;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    public SEMSPortalHandler(Thing thing, HttpClientFactory httpClientFactory) {
+        super(thing);
+        httpClient = httpClientFactory.getCommonHttpClient();
+        gson = new GsonBuilder().setDateFormat(SEMSPortalBindingConstants.DATE_FORMAT).create();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!loggedIn) {
+            logger.debug("Not logged in. Ignoring command {} for channel {}", command, channelUID);
+            return;
+        }
+        if (command instanceof RefreshType) {
+            scheduler.execute(() -> {
+                updateChannelState(channelUID);
+            });
+        }
+    }
+
+    private void updateChannelState(ChannelUID channelUID) {
+        if (!loggedIn) {
+            logger.debug("Not logged in. Ignoring update request for channel {}", channelUID.getAsString());
+            return;
+        }
+        ensureRecentStatus();
+        if (CHANNEL_CURRENT_OUTPUT.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getCurrentOutput(currentStatus));
+        } else if (CHANNEL_TODAY_TOTAL.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getDayTotal(currentStatus));
+        } else if (CHANNEL_MONTH_TOTAL.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getMonthTotal(currentStatus));
+        } else if (CHANNEL_OVERALL_TOTAL.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getOverallTotal(currentStatus));
+        } else if (CHANNEL_ONLINE_STATE.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getOnline(currentStatus));
+            if (currentStatus != null && currentStatus.isOnline() != stationOnline) {
+                stationOnline = currentStatus.isOnline();
+                triggerChannel(channelUID, OnOffType.from(stationOnline).name());
+            }
+        } else if (CHANNEL_TODAY_INCOME.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getDayIncome(currentStatus));
+        } else if (CHANNEL_TOTAL_INCOME.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getTotalIncome(currentStatus));
+        } else if (CHANNEL_LASTUPDATE.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getLastUpdate(currentStatus));
+        }
```

Comment:
That is a lot of else if statements, IMHO this should be https://www.w3schools.com/java/java_switch.asp

Reply:
Wow, that made me feel old. I thought you couldn't with Strings but seems it was added back in java 1.7...